### DISCUSSION TITLE
[Docs] Confusion about Volume-Servers and Volume-Files when talking about ‚Volumes‘

### DISCUSSION THREAD

**Discussion by bjesuiter** [seaweedfs/seaweedfs#2495] [Q&A]
I'm trying to come up with a system to use seaweed fs as my home nas server. The exact case and my current understanding can be found at the bottom of this issue. 

Now I try to grasp the concept of replication between volume-servers. 
The problem for me is, that the usage of 'Volume' in the docs seems to be switching between meaning 'Volume-Files inside a Volume Server' ans 'one or more Volume Servers' 

Additionally comes the confusion, that the term 'Volume' is traditionally associated with direct HDDs or SSDs. 
I think we, as a community, should clarify this somehow in the docs. 
If I find a working solution for my setup, I'll happily contribute my example case to the docs to help future readers.

## My Current Setup

I have a 4-Bay Storage Appliance with 2x4TB HDDs and 2x 6TB HDDs. 
Each pair is mirrored in ZFS and combined to a pool in ZFS, of around 8 TB. 

## My SeaweedFS Plan with Question: Is this understanding correct? : 

I found the following 

1. weed master has Param "volumeSizeLimit" 
	1. = max  size of volume files 
	      (can be larger than 30GB because of large_disk version of seaweed FS supporting volume greater than 30GB)
2. weed volume has Param max number of volumes 
	1. = Max number of volume files 
	2. Set so that HDDs do not overfill 

=> Therefore I need 4 Volume-Servers (1 per HDD), configured like this: 
     1. Per 4 TB HDD : 50 GB per Volume-File * 80 Volumes 
     2. Per 6 TB HDD: 50 GB per Volume-File * 120 Volumes 
     
=> **Replication Setup: 001 - Questions**
- Is it a problem, that two of the underlying HDDs have greater capacity???
- Is this the correct mental model or is it planned like this, that each HDD or SSD needs it's own volume server?

The docs say for **Fixing Replication** 

> If one replica is missing, there are no automatic repair right away. This is to prevent over replication due to transient volume sever failures or disconnections. In stead, the volume will just become readonly. For any new writes, just assign a different file id to a different volume.

**Multiple Questions** 
- `If one Replica is missing` => Does this mean a missing HDD in one volume server or a missing Volume file?
- `Instead, the volume will just become readonly` => Again: Does this mean a specific volume file or a volume server?

**Comment by chrislusf** [comment]
> the term 'Volume' is traditionally associated with direct HDDs or SSDs.

Not used this way. This was the source of confusion. Some readers simply have the wrong impression that SeaweedFS can not support files larger than 30GB, because each volume is default to 30GB. I need some help to convey this concept better in the wiki.

The word `volume` is always the volume file. `volume server` is the server. 

> * Is it a problem, that two of the underlying HDDs have greater capacity???

This is fine.

> * Is this the correct mental model or is it planned like this, that each HDD or SSD needs it's own volume server?

This is correct. 

Also, for convenience, One volume server can support multiple directories.

> * `If one Replica is missing` => Does this mean a missing HDD in one volume server or a missing Volume file?

One missing volume file.

> * `Instead, the volume will just become readonly` => Again: Does this mean a specific volume file or a volume server?

One specific volume file.
